instrumentor/controllers/instrumentationconfig/instrumentedapplication_controller.go

```diff
@@ -55,19 +56,18 @@ func (r *InstrumentedApplicationReconciler) Reconcile(ctx context.Context, req c
 
 	instrumentationRules := &odigosv1.InstrumentationRuleList{}
 	err = r.Client.List(ctx, instrumentationRules)
+	if client.IgnoreNotFound(err) != nil {
```

Comment:
can a list operation return not found? or is it only for safety?

If it's not found, don't we want to return and not cause any updates?

Reply:
Thich check is for errors which are not `NotFound`